README.md

```diff
@@ -3,7 +3,77 @@ Legends of the Wulin Character Sheet
+Instead, LotWCS will provide an import/export feature for you to store your characters.
+
+# Technical stuff
+Here we will bore you with technical considerations, for developers of LotWCS mainly.
+
+## App Model
+This application uses the standard React-Redux pattern.
+The whole application relies on a Redux Store to store the character information.
+The character evolves from an initial state to the current state through user actions.
+
+The Redux architecture handle state evolutions by applying a list of actions to the current state, returning a new updated state.
+The Redux Store handle the dispatching for actions (from user inputs mainly) to the reducers.
+
+The state is used to feed React Components to display the character.
+React Components are subdivided by responsibility: container Components are there to link the store to render Components.
+
+## File hierarchy
+The src directory is divided by responsibility:
+
+### actions
+Here are the action creators, a lot of helper functions present to generate an input to a specific action understandable by the reducers.
+
+### components
+__This directory regroups "renderer" components, handling only rendering and catching user inputs.__
+
+Theirs props must provide everything they need to know to do their job.
+
+### constants
+Constants definitions, mainly used to specify available action.type value, on which the reducers rely to identify actions.
+
+### containers
+__This directory regroups "containers" components and other functions for reading elements from the state.__
+
+Those are obtained by connecting the store and the state to a "renderer" component, following [the canonical method from Redux](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options):
+* mapStateToProps: maps the state to the props, by transforming it, by adding information from static data ;
+* mapDispatchToProps: maps dispach actions to the props, passing through the props the dispatch+action creator, allowing subsequent "renderer" components to trigger actions.
+* mergeProps: merges both previous "sub props" to the one sent to the "renderer component".
+Each container component is connected to a renderer component of the same name.
+
+### data
+Static data and some data extraction functions.
+
+### reducers
+__This directory regroups reducers, performing modifying actions to the state.__
+
+Here are the global reducer, the one plugged to the Redux Store and a lot of sub reducers, handling a specific subset of actions.
+One important rule to note is that no action is applied to the state by different reducers.
+
+### types
+Typescript types definitions.
+
+## Data flow
+The data trickle down from the state to feed the props of the components.
+
+When a container maps the state to the props of a component, it must check which actions are available or not, and it must take prepare the cost of the action by checking for discount.
```

Comment:
It's not really clear you're writing about skills costs and not js performance cost of the action.
You might want to write a short description before to explain why cost management is such a pain and here explaining your technical strategy as you will explain it in your next reactJs/humanTalks meetup :)